File: DiscussionForum-Angular/Models/Question.cs
using System;
using System.Text.Json.Serialization;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DiscussionForum_Angular.Models;

public class Question
{

    [Key]
    public int QuestionId { get; set; }

    public string Title { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;
    public DateTime Created { get; set; } = DateTime.Now;

    [ForeignKey("User")]
    public string Id { get; set; } = string.Empty;

    public virtual List<Reply>? Replies { get; set; }
    public virtual User? User { get; set; } = default!;
}

File: DiscussionForum-Angular/ClientApp/src/app/questions/question.ts
import { IUser } from "../users/user"
import { IReply } from "../replies/reply"

export interface IQuestion {
  questionId: number;
  id: string;
  title: string;
  content: string;
  created: Date;
  user: IUser;
  replies: IReply[];
}
